ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/sync/backwardsync/BackwardChain.java

```diff
@@ -179,7 +179,7 @@ public synchronized void addNewHash(final Hash newBlockHash) {
   }
 
   public synchronized Optional<Hash> getFirstHashToAppend() {
-    return Optional.ofNullable(hashesToAppend.peek());
+    return Optional.ofNullable(hashesToAppend.poll());
```

Comment:
Was this change added by mistake?

Reply:
Yes, thanks for catching it. It was accidentally included from a merge upstream.